Fix Cabana configure error in CI

zackgalbreath opened the pull request:
After #1382 was merged, Cabana began failing to find Kokkos during the configure step.

Fix this by making Cabana unconditionally depend on Kokkos.

eugeneswalker reviewed (dismissed):
Thanks!

kwryankrattiger commented on stacks/e4s-cray-rhel/spack.yaml:
```diff
@@ -98,7 +98,7 @@ spack:
   - argobots
   - boost +python +filesystem +iostreams +system
-  - cabana
+  - cabana +openmp
```
Constrain this at the packages config level for kokkos with for all of these stacks. Cabana will detect this correctly then.

```
kokkos:
  prefer: ["+serial +openmp"]
```

kwryankrattiger reviewed (commented):
This also needs a fix in the cabana recipe to always depend on kokkos. Kokkos enforces the requirement of at least one backend being specified, so the constraint in the stacks should be on kokkos instead of cabana.

```
depends_on("kokkos")
```

kwryankrattiger commented on repos/spack_repo/builtin/packages/cabana/package.py:
```diff
@@ -51,6 +51,8 @@ class Cabana(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("c", type="build", when="+mpi")
     depends_on("cxx", type="build")
 
+    depends_on("kokkos")
```
small nit, but can you put this right above the loops for the kokkos variant propagation to keep them together

zackgalbreath committed:
Fix Cabana configure error in CI

After #1382 was merged, Cabana began failing to find Kokkos during
the configure step.

Fix this by making Cabana unconditionally depend on Kokkos.
